### PR TITLE
Adds support for native Typescript enums

### LINKED ISSUE #35: Circular dependency error for enum types
## Bug description

_This could be considered a feature request rather than a bug since I haven't seen any mention of enum handling in the codebase 🤷_ 

When generating a zod validator from a type which references an enum, the generation produces a waring on the CLI and an empty generated file.

```
 ›   Warning: Some schemas can't be generated due to circular dependencies:
 ›   tagSchema
✔ Validating generated types
🎉 Zod schemas generated!
✨  Done in 1.99s.
```
### Input

```ts

export enum TagName {
	Button = 'button'
};

export type Tag = {
	name: TagName,
}

```

### Expected output

```ts
// Generated by ts-to-zod
import { z } from 'zod'
import { TagName } from '../types/tag-simple';

export const tagNameSchema = z.nativeEnum(TagName);

export const tagSchema = z.object({
  name: tagNameSchema
})

```

### Actual output

```ts

```

**Versions**

- Typescript: `v4.3.4`
- Zod: `v3.1.0`

### PR DESCRIPTION
**Pull request by anglinb**
# Why

Fixes #35 by adding support for native typescript enums

### CODE REVIEW

**Review by fabien0102 — CHANGES_REQUESTED** [review]
Let's just remove this commented line and we are good to go, thanks for this addition!

**Comment by fabien0102** [comment]
Let's also add a little test into `generateZodSchema.test.ts`

**Comment by anglinb** [comment]
Closing in favor of https://github.com/fabien0102/ts-to-zod/pull/40. Handled the feedback on the other pr.

**Comment by anglinb** [comment]
> Let's also add a little test into `generateZodSchema.test.ts`

https://github.com/fabien0102/ts-to-zod/pull/40/commits/7de337204eb25d998d7e5d8be2b4a452589811c3

File: src/core/generateZodSchema.ts [hunk 1]
```diff
@@ -57,6 +57,8 @@ export function generateZodSchemaVariableStatement({
 }: GenerateZodSchemaProps) {
   let schema: ts.CallExpression | ts.Identifier | undefined;
   const dependencies: string[] = [];
+  let requiresImport = false;
+  // const imports: string[] = [];
```
[the remark this answers is not in the document]
Comment (anglinb) [reply src/core/generateZodSchema.ts:61]: https://github.com/fabien0102/ts-to-zod/pull/40/commits/bf1a203c346c421bf472384b5151a4975561e1fc

**Pull request closed without merging**